config-model/src/main/java/com/yahoo/vespa/model/admin/otel/OpenTelemetryConfigGenerator.java

```diff
@@ -203,4 +251,68 @@ void addStatePorts(List<StatePortInfo> portList) {
     List<String> referencedPaths() {
         return List.of(ca_file, cert_file, key_file);
     }
+
+    private String zoneAttr() {
+        if (zone == null) return "noZone";
+        return zone.environment().value() + "." + zone.region().value();
+    }
+    private String appIdAttr() {
+        if (applicationId == null) return "noApp";
+        return applicationId.toFullString();
+    }
+    private String systemAttr() {
+        if (zone == null) return "noSystem";
+        return zone.system().value();
+    }
+    private String tenantAttr() {
+        if (applicationId == null) return "noTenant";
+        return applicationId.tenant().value();
+    }
+    private String appNameAttr() {
+        if (applicationId == null) return "noAppName";
+        return applicationId.application().value();
+    }
+    private String appInstanceAttr() {
+        if (applicationId == null) return "noAppInstance";
+        return applicationId.instance().value();
+    }
+    private String cloudAttr() {
+        if (zone == null) return "noCloud";
+        return zone.cloud().name().value();
+    }
```

Comment:
Maybe we should add the dimensions iff the value is not null?

Reply:
done, please take another look